### ISSUE TITLE
Chrome Deprecated Synchronous XMLHttpRequest

### ISSUE THREAD

**Issue by williamfligor** [skulpt/skulpt#514]
When importing a custom lib, this warning occurs

> Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check http://xhr.spec.whatwg.org/.

I believe this comes from Sk.loadExternalLibraryInternal_ in import.js

**Comment by waywaaard** [comment]
It is caused by this function. We need to assure that we do not continue compiling before loading the external modules. Maybe we should refactor this and use Promises?

**Comment by albertjan** [comment]
@meredydd would is be possible to "halt" execution till the file has been resolved via a promise?

**Comment by meredydd** [comment]
Yes - this is already possible when importing Python modules - the user-supplied loading functions are allowed to suspend. (It's a little ugly, as it was one of the first things written, before we had Sk.misceval.chain and similar - the whole thing could do with a refactor.) 

What's the context in which loadExternalLibrary is called? We just need to Suspensionise it.
